### PR TITLE
ci: github acceptance improvements [WPB-8666] [WPB-9133]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8666" title="WPB-8666" target="_blank">TaskWPB-8666</a>  [Android] Setup pipeline step to run acceptance tests on CI
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- Run acceptanceTests on GitHub Actions, only for develop, release/candidate, prod , internal and main or PR targeting those branches
- Preserve logic to archive tests results with commit hash

### CODE REVIEW

File: .github/workflows/gradle-run-ui-tests.yml [hunk 1]
```diff
@@ -61,9 +73,27 @@ jobs:
...
+        id: get_commit_hash
+        run: echo "::set-output name=commit_hash::$(git rev-parse HEAD)"
+
+      - name: Archive test results
+        run: |
+          zip -r integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}.zip app/build/reports/androidTests/connected/debug/flavors/${{ github.event.inputs.FLAVOR }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}
+          path: integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}.zip
```
Comment (yamilmedina) [inline .github/workflows/gradle-run-ui-tests.yml]: Awesome 👏  ! I think we forgot to have it for the `unitTests` too so I'll create a task for it, but this is what we need 🙌

**Review by yamilmedina — APPROVED** [review]
Looking pretty good, added a reminder for the hash part of archiving

**Comment by AndroidBob** [comment]
Build [4756](https://10.10.124.134/job/AR-build-pipeline/4756/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9680660)(9680660)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9680660)(9680660)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4761](https://10.10.124.134/job/AR-build-pipeline/4761/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9681807)(9681807)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9681807)(9681807)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4771](https://10.10.124.134/job/AR-build-pipeline/4771/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9687081)(9687081)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9687081)(9687081)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4775](https://10.10.124.134/job/AR-build-pipeline/4775/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9687914)(9687914)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9687914)(9687914)-staging-compat.apk)

File: .editorconfig [hunk 2]
```diff
@@ -16,6 +16,11 @@ ij_smart_tabs = false
 ij_visual_guides = none
 ij_wrap_on_typing = false
 
+# Specific settings for YAML files
+[{*.yml,*.yaml}]
+indent_style = space
+indent_size = 2
+
```
Comment (alexandreferris) [inline .editorconfig:23]: 🤔 tabs vs spaces? 🤔

**Review by alexandreferris — APPROVED** [review]

File: app/src/test/kotlin/com/wire/android/ui/CallActivityViewModelTest.kt [hunk 3]
```diff
@@ -87,6 +88,7 @@ class CallActivityViewModelTest {
                 .arrange()
 
             viewModel.switchAccountIfNeeded(userId)
+            advanceUntilIdle()
```
Comment (yamilmedina) [inline app/src/test/kotlin/com/wire/android/ui/CallActivityViewModelTest.kt:91]: good catch

File: .editorconfig (hunk 2 above)
Comment (Garzas) [reply .editorconfig:23]: I added this code style because current scripts are formatted in that way 😅

**Review by yamilmedina — APPROVED** [review]
Looks good to me, I checked it out the artifacts generated and everything in order 🚀

**Comment by AndroidBob** [comment]
Build [4786](https://10.10.124.134/job/AR-build-pipeline/4786/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9690193)(9690193)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9690193)(9690193)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4789](https://10.10.124.134/job/AR-build-pipeline/4789/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9697051)(9697051)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9697051)(9697051)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4796](https://10.10.124.134/job/AR-build-pipeline/4796/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9697890)(9697890)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9697890)(9697890)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4803](https://10.10.124.134/job/AR-build-pipeline/4803/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9721891)(9721891)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9721891)(9721891)-staging-compat.apk)

**Pull request merged**